src/components/color-picker/color-picker.tsx

```diff
@@ -0,0 +1,91 @@
+import { Component, h, State } from '@stencil/core';
+
+/**
+ * This component enables you to select a swatch from out color palette, simply
+ * by clicking on it. You can then copy the css variable name of the chosen color
+ * and use it where desired.
+ *
+ * The color picker can also show you a preview of any valid color name or color value.
+ *
+ * @exampleComponent limel-example-color-picker
+ */
+
+@Component({
+    tag: 'limel-color-picker',
+    shadow: true,
+    styleUrl: 'color-picker.scss',
+})
+export class ColorPicker {
+    /**
+     * Name or code of the chosen color
+     */
+    @State()
+    public color: string;
+
+    @State()
+    private isOpen = false;
+
+    public render() {
+        const background = this.color ? { '--background': this.color } : {};
+
+        return (
+            <div class="color-picker">
+                <limel-popover open={this.isOpen} onClose={this.onPopoverClose}>
+                    <div
+                        class="picker-trigger"
+                        slot="trigger"
+                        style={background}
+                        role="button"
+                        tabindex="0"
+                        onClick={this.openPopover}
+                        title="Click to pick a color"
+                    />
+                    <limel-color-picker-palette
+                        searchedColor={this.color}
+                        onChange={this.onChosenColor}
+                    />
+                </limel-popover>
+
+                <limel-input-field
+                    label="Chosen color"
+                    helperText="You can also type a color name or value to preview it here"
+                    value={this.color}
+                    onChange={this.handleChange}
+                />
+            </div>
+        );
+    }
+    private openPopover = (event: MouseEvent) => {
+        event.stopPropagation();
+        this.isOpen = true;
+    };
+
+    private onPopoverClose = (event: CustomEvent) => {
+        event.stopPropagation();
+        this.isOpen = false;
+    };
+
+    private onChosenColor = (event: any) => {
+        this.color = event.detail;
+    };
+
+    private handleChange = (event: CustomEvent<string>) => {
+        this.color = event.detail;
+        const pickerPalette = this.getPickerPallete();
+        this.removeSelectedColor(pickerPalette);
+    };
+
+    private getPickerPallete() {
+        return document
+            .querySelector('limel-popover-surface')
+            .shadowRoot.querySelector('limel-color-picker-palette');
+    }
+
+    private removeSelectedColor(
+        pickerPalette: HTMLLimelColorPickerPaletteElement
+    ) {
+        pickerPalette.shadowRoot
+            .querySelector('.selected-color-palette')
+            ?.classList.remove('selected-color-palette');
+    }
```

Comment:
Do we really need to manipulate the DOM manually like this? Can we not handle everything in the `render` method? :thinking:

Reply:
Maybe, but I'm not sure how.. , I'm manipulating with the DOM since it was necessary to dynamically add CSS on click

Reply:
We should use the current value to determine if one of the colors is selected or not :slightly_smiling_face: